test/sync/src/androidTest/kotlin/io/realm/kotlin/test/shared/SyncedRealmTests.kt

```diff
@@ -231,6 +232,42 @@ class SyncedRealmTests {
         realm3.close()
     }
 
+    @Test
+    fun errorHandlerReceivesPermissionDeniedSyncError() {
+        val channel = Channel<SyncException>(1).freeze()
+        val (email, password) = randomEmail() to "password1234"
+        val user = runBlocking {
+            app.createUserAndLogIn(email, password)
+        }
+
+        val config = SyncConfiguration.Builder(
+            schema = setOf(ParentPk::class, ChildPk::class),
+            user = user,
+            partitionValue = partitionValue
+        ).errorHandler { _, error ->
+            channel.trySend(error)
+        }.build()
+
+        // Remove permissions to generate a sync error containing ONLY the original path
+        // This way we assert we don't read wrong data from the user_info field
+        runBlocking {
+            app.asTestApp.changeSyncPermissions(SyncPermissions(read = false, write = false)) {
+                runBlocking {
+                    val deferred = async { Realm.open(config) }
+
+                    val error = channel.receive()
+                    val message = error.message
```

Comment:
According to James:
> This flag is controlled by the server. I can’t recall the reasons for this so if you need further explanation, best to ask the server team. In terms of the client’s behaviour, even though the error is not “fatal” the client will disconnect sync, and transition to the “inactive” state.

cc. @nhachicha

Reply:
I think we still need to assert the type of exception and make sure it belongs to the expected category, despite Core not setting the flag accordingly ...

Reply:
Now our sync error mapper returns `UnrecoverableSyncException` if we detect a permission denied error. At least users can now see the error returned in the callback is unrecoverable so that they can act upon it.